octopusdeploy/lifecycles.go

```diff
@@ -0,0 +1,204 @@
+package octopusdeploy
+
+import (
+	"fmt"
+	"gopkg.in/go-playground/validator.v9"
+	"net/url"
+
+	"github.com/dghubble/sling"
+)
+
+type LifecycleService struct {
+	sling *sling.Sling
+}
+
+func NewLifecycleService(sling *sling.Sling) *LifecycleService {
+	return &LifecycleService{
+		sling: sling,
+	}
+}
+
+type Lifecycles struct {
+	Items []Lifecycle `json:"Items"`
+	PagedResults
+}
+
+type Lifecycle struct {
+	ID                      string          `json:"Id,omitempty"`
+	Name                    string          `json:"Name" validate:"required"`
+	Description             string          `json:"Description,omitempty"`
+	ReleaseRetentionPolicy  RetentionPeriod `json:"ReleaseRetentionPolicy"`
+	TentacleRetentionPolicy RetentionPeriod `json:"TentacleRetentionPolicy"`
+	Phases                  []Phase         `json:"Phases"`
+}
+
+type RetentionUnit string
+
+const (
+	RetentionUnit_Days = "Days"
+	RetentionUnit_Items = "Items"
+)
+
+type RetentionPeriod struct {
+	Unit              RetentionUnit `json:"Unit"`
+	QuantityToKeep    int32         `json:"QuantityToKeep"`
+	ShouldKeepForever bool          `json:"ShouldKeepForever"`
+}
+
+type Phase struct {
+	ID                                 string          `json:"Id,omitempty"`
+	Name                               string          `json:"Name" validate:"required"`
+	MinimumEnvironmentsBeforePromotion int32           `json:"MinimumEnvironmentsBeforePromotion"`
+	IsOptionalPhase                    bool            `json:"IsOptionalPhase"`
+	ReleaseRetentionPolicy             *RetentionPeriod `json:"ReleaseRetentionPolicy"`
```

Comment:
This property is optional, I guessed the `*` is the right way to do this

Reply:
You can use `omitempty` on the json tag like here: https://github.com/MattHodge/go-octopusdeploy/blob/master/octopusdeploy/machines.go#L40